dump/main.py

```diff
@@ -0,0 +1,233 @@
+@click.argument('identifier', required=True, type=str) 
+@click.option('--show', '-s', is_flag=True, default=False, help='Display Modules Available', is_eager=True, expose_value=False, callback=show_modules)
+@click.option('--all-modules', '-a', is_flag=True, default=False, help='Print modules in a techsupport-friendly format', is_eager=True, expose_value=False, callback=all_modules)
+@click.option('--db', '-d', multiple=True, help='Only dump from these Databases')
+@click.option('--table', '-t', is_flag=True, default=False, help='Print in tabular format', show_default=True)
+@click.option('--key-map', '-k', is_flag=True, default=False, help="Only fetch the keys matched, don't extract field-value dumps", show_default=True)
+@click.option('--verbose', '-v', is_flag=True, default=False, help="Prints any intermediate output to stdout useful for dev & troubleshooting", show_default=True)
+@click.option('--namespace', '-n', default=DEFAULT_NAMESPACE, type=str, show_default=True, help='Dump the redis-state for this namespace.')  
+def state(ctx, module, identifier, db, table, key_map, verbose, namespace):
+    """
+    Dump the redis-state of the identifier for the module specified
+    """
+    if not multi_asic.is_multi_asic() and namespace != DEFAULT_NAMESPACE:
+        click.echo("Namespace option is not valid for a single-ASIC device")
+        ctx.exit()
+    
+    if multi_asic.is_multi_asic() and (namespace != DEFAULT_NAMESPACE and namespace not in multi_asic_ns_choices()):
+        click.echo("Namespace option is not valid. Choose one of {}".format(multi_asic_ns_choices()))
+        ctx.exit()
+    
+    if module not in plugins.dump_modules:
+        click.echo("No Matching Plugin has been Implemented")
+        ctx.exit()
+    
+    if verbose:
+        os.environ["VERBOSE"] = "1"
+    else:
+        os.environ["VERBOSE"] = "0"
+    
+    ctx.module = module
+    obj = plugins.dump_modules[module]()
+    
+    if identifier == "all":
+        ids = obj.get_all_args(namespace)
+    else:
+        ids = identifier.split(",")
+        
+    params = {}
+    collected_info = {}
+    params['namespace'] = namespace
+    for arg in ids: 
+        params[plugins.dump_modules[module].ARG_NAME] = arg 
+        collected_info[arg] = obj.execute(params)
+        
+    if len(db) > 0:
+        collected_info = filter_out_dbs(db, collected_info)
+    
+    vidtorid = extract_rid(collected_info, namespace)
+    
+    if not key_map:
+        collected_info = populate_fv(collected_info, module, namespace)
+    
+    for id in vidtorid.keys():
+        if  vidtorid[id] and vidtorid:
+            collected_info[id]["ASIC_DB"]["vidtorid"] = vidtorid[id]
+         
+    print_dump(collected_info, table, module, identifier, key_map)
+    
+    return 
+
+def extract_rid(info, ns):
+    r = SonicV2Connector(namespace=ns, host="127.0.0.1")
+    r.connect("ASIC_DB")
+    vidtorid = {}
+    for arg in info.keys():
```

Comment:
Can you make sure you fill the arg only when the get_v_r_map returns a non empty object? In this way we can avoid the if condition in the for loop above